packages/lit-html/src/lib/lit-html.ts

```diff
@@ -463,6 +472,12 @@ class TemplateInstance {
     this.__template = template;
   }
 
+  __disconnect() {
```

Comment:
Let's add a comment about this being a part tree walk and a TODO for perf testing.